pkg/clustermanager/kubeconfig_manager.go

```diff
@@ -0,0 +1,61 @@
+package clustermanager
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/providers"
+	"github.com/aws/eks-anywhere/pkg/retrier"
+)
+
+type KubeconfigManager struct {
```

Comment:
Could we find a different name? manager is an overloaded term
plus it doesn't tell us a lot

maybe `KubeconfigReader`?